src/main/java/dao/CsvDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package dao;
+
+import static db.StorageTransaction.fruitTransactionList;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import service.impl.PositiveBalanceCheck;
+
+public class CsvDaoImpl implements CsvDao {
+    @Override
+    public List<String> readFromFile(String filename) {
+        List<String> dataFromFile;
+        try {
+            dataFromFile = Files.readAllLines(Path.of(filename));
+        } catch (Exception e) {
+            throw new RuntimeException("Can`t read from file " + filename, e);
+        }
+        return dataFromFile;
+    }
+
+    @Override //              TODO positive balance check !!!
+    public void writeToFile(Map<String, Integer> fruitTotalBalanceMap, String toFileName) {
+        if (!PositiveBalanceCheck.check(fruitTotalBalanceMap)) {
+            throw new RuntimeException("The fruit storage balance is negative!");
+        }
```

Comment:
lets throw an exception when performing substraction in a handler, it whould signal that there is not enough fruits in strrage